test/gtest/ucp/test_ucp_tag_fallback.cc

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+ *
+ */
+#include "ucp_test.h"
+#include <ucs/sys/sys.h>
+
+#include <sys/mman.h>
+
+#define MSG_SIZE  (4 * 1024 * ucs_get_page_size())
+
+class test_ucp_tag_fallback : public ucp_test {
+public:
+    void init() {
+        /* forbid zcopy access because it will always fail due to read-only
+         * memory pages (will fail to register memory) */
+        modify_config("ZCOPY_THRESH", "inf");
+        ucp_test::init();
+        sender().connect(&receiver(), get_ep_params());
+        receiver().connect(&sender(), get_ep_params());
+    }
+
+    static ucp_params_t get_ctx_params() {
+        ucp_params_t params = ucp_test::get_ctx_params();
+        params.field_mask  |= UCP_PARAM_FIELD_FEATURES;
+        params.features     = UCP_FEATURE_TAG;
+        return params;
+    }
+};
+
+UCS_TEST_P(test_ucp_tag_fallback, fallback)
+{
+    void *send_buffer;
+    void *recv_buffer;
+    ucs_status_ptr_t send_req;
+    ucs_status_ptr_t recv_req;
```

Comment:
can use inline declaration for all these vars, e.g
`void *send_buffer = mmap(... `